org.eclipse.sirius.emfjson/src/main/java/org/eclipse/sirius/emfjson/resource/JsonResource.java

```diff
@@ -565,6 +565,21 @@ interface IJsonResourceProcessor {
          */
         Object getValue(EObject eObject, EStructuralFeature feature, Object value);
 
+        /**
+         * Called during the parsing of the JsonResource (at loading time). Use this method to change the URI of a value from
+         * a non-containment reference in order to change the value of a proxy. Return the given uri directly if you don't want
+         * to perform any change.
+         *
+         * @param eObject
+         *            the object containing the reference
+         * @param eReference
+         *            the reference to set
+         * @param uri
+         *            the uri of the object to add to the reference
+         * @return The new value of the uri to use
+         */
+        String getEObjectUri(EObject eObject, EReference eReference, String uri);
```

Comment:
Technically this breaks API as is add a new method that needs to be implemented by all existing implementers of this interface. I guess the implementation in NoOp below could be put directly as a default in the interface itself, which would make the change non-breaking.